### PR TITLE
added Tax start date page

### PR DESCRIPTION
**Pull request by mauro-fanton**
added more tests and save and continues button

linked pages to the TaxStartDate page

removed unused import

saved start date to db

purifing liablityDetails when save and sontinues on both exceeded and expect to exceed threshold weight pages

renamed

### Description of Work carried through

Brief description of what/why/how.

#### Check list 
 - [x] `./precheck` was executed (Integration/Component/Unit tests)
 - [x] `sbt scalafmt test:scalafmt` was executed
 - [ ] Required Environment Config has been amended/added - N/A
 - [x] Links to dependencies have been included (BE/FE work)
 - [x] User Acceptance Tests (UAT) were run locally and have passed.
 - [ ] No Accessibility errors/warnings reported by Wave - N/A

### CODE REVIEW

File: app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/liability/TaxStartDateController.scala [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+@Singleton
+class TaxStartDateController @Inject() (
+  authenticate: AuthAction,
+  journeyAction: JourneyAction,
+  taxStarDateService: TaxStartDateService,
+  mcc: MessagesControllerComponents,
+  page: tax_start_date_page,
+  override val registrationConnector: RegistrationConnector
+)(implicit ec: ExecutionContext)
+    extends LiabilityController(mcc) with Cacheable with I18nSupport {
```
Comment (kiranprakashdigital) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/liability/TaxStartDateController.scala:42]: Can we change to FrontendController

File: app/uk/gov/hmrc/plasticpackagingtax/registration/views/liability/tax_start_date_page.scala.html [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    @sectionHeader(messages("liability.sectionHeader"))
+    @pageHeading(messages("liability.taxStartDate.pageHeading"))
+    @paragraphBody(messages("liability.taxStartDate.hint1",
+                                startDate.getDayOfMonth + " " +
+                                messages(s"date.month.${startDate.getMonthValue}") + " " +
+                                startDate.getYear ))
+    @paragraphBody(
+        if(hasExceededThresholdWeight)
+            messages("liability.taxStartDate.ThreshHoldBreached.hint")
```
Comment (kiranprakashdigital) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/views/liability/tax_start_date_page.scala.html:46]: `ThreshHoldBreached` should be `thresholdBreached`

**Review by kiranprakashdigital — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/liability/TaxStartDateController.scala (hunk 1 above)
Comment (mauro-fanton) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/liability/TaxStartDateController.scala]: Yes, true, I do not need to extends LiabilityController as I do not use anything from it. Thanks for pointing this out. Done

File: app/uk/gov/hmrc/plasticpackagingtax/registration/views/liability/tax_start_date_page.scala.html (hunk 2 above)
Comment (mauro-fanton) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/views/liability/tax_start_date_page.scala.html]: yep. I missed that. Thanks.
Done

**Review by kiranprakashdigital — APPROVED** [review]

**Pull request merged**